### ISSUE TITLE
Test ARM32 tailcalls for profiler

### ISSUE THREAD

**Issue by sywhang** [dotnet/coreclr#14724]
Profiler ELT hooks for ARM32 tailcall has not been tested yet, because JIT optimizations for tailcall is missing on ARM32. The tailcall hook for arm32 has already been added by @rartemev with #14713, so once JIT optimizations are added in, we need to test this hook.

**Comment by sywhang** [comment]
@rartemev @davmason @lt72 FYI

**Comment by sywhang** [comment]
Per discussion with @lt72 and @RussKeldorph on email, since fast tailcall for ARM32 is optional for 2.1 and will probably not make it in time for 2.1, we will move this to Future.

**Comment by BruceForstall** [comment]
@sywhang Is this issue still valid?

JIT fast tailcall is an optimization, and should not block testing ARM32 ELT hooks for profiler. These hooks should function, and should be tested for ARM32.

When fast tailcall optimization is implemented, it will also require ELT hook testing. But that can happen later, when the optimization is implemented.

**Comment by sywhang** [comment]
@BruceForstall Yes, the ARM32 ELT hooks have already been tested, except for just the fast tailcall path. That's the only path we haven't been tested and this issue was made to track testing fast tailcall optimization when it is implemented. I guess we could update the title to say "fast tailcall" instead of just tailcall.
